### ISSUE TITLE
Uncaught Error: EBADF, bad file descriptor

### ISSUE THREAD

**Issue by NoelAbrahams** [TypeStrong/atom-typescript#33]
1. Install atom-typescript.
2. Error occurs soon after (x4)

**Atom Version**: 0.176.0
**System**: Microsoft Windows 8.1 Pro
**Thrown From**: [atom-typescript](https://github.com/TypeStrong/atom-typescript) package, v0.9.0

### Stack Trace

Uncaught Error: EBADF, bad file descriptor

```
At fs.js:559

Error: EBADF, bad file descriptor
  at Error (native)
  at Object.fs.writeSync (fs.js:559:20)
  at SyncWriteStream.write (fs.js:1768:6)
  at Socket.<anonymous> (c:\Users\Noel\.atom\packages\atom-typescript\node_modules\atom-package-dependencies\node_modules\shelljs\src\exec.js:113:22)
  at Socket.emit (events.js:129:20)
  at readableAddChunk (_stream_readable.js:161:16)
  at Socket.Readable.push (_stream_readable.js:125:10)
  at Pipe.onread (net.js:522:20)

```

### Commands

```
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "autocomplete"
    ],
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
atom-typescript, v0.9.0

# Dev
No dev packages
```

/cc @atom/core

------------ 

# Workaround

Just restart atom after these errors. They only happen *once* when we try to install `linter` and `autocomplete-plus` on your behalf.

**Comment by basarat** [comment]
@NoelAbrahams Thanks! this is because of my trying to install `autocomplete-plus` on your behalf. 

Few things would help me debug it better:
* can you run `apm` on the command line i.e. is it in your system `path`
* are you running with special permissions like `administrator` by any chance? (shouldn't be).

**Comment by kiyokura** [comment]
I got the same error.

I tried that :
- uninstall atom-typescript
- open cmd as administrator.
- execute command :  apm install atom-typescript


However, the same error occurs.

**Comment by basarat** [comment]
Workaround is :
```shell
apm install autocomplete-plus
apm install atom-typescript
```

**Comment by basarat** [comment]
I can reproduce it. Great. Sorry for that.

**Comment by kiyokura** [comment]
The workaround worked well for me.
thx!

**Comment by basarat** [comment]
Updated workaround: Just restart atom after these errors. They only happen once when we try to install linter and autocomplete-plus on your behalf.

**Comment by basarat** [comment]
This is the source of the error: https://github.com/travs/atom-package-dependencies/issues/6 I already fixed it  https://github.com/travs/atom-package-dependencies/pull/7 but looking at `atom-package-dependencies/index.js` on NPM it doesn't seem to have my fix. Pinging @travs

**Comment by basarat** [comment]
Fixed in `atom-typescript@v0.11.0` 

Fix commit : https://github.com/TypeStrong/atom-typescript/commit/a210f26312318059d59ed4f81820bffbcec474d5

**Comment by kiyokura** [comment]
:thumbsup:
